src/components/utils/detectSystem.js

```diff
@@ -0,0 +1,84 @@
+const SYSTEMS_MAPPING = {
+  os: [
+    {
+      name: 'Windows Phone',
+      codeName: 'Windows Phone',
```

Comment:
Is `codeName` a nickname?

Reply:
The exact syntax of user agents is `<product> / <product-version> <comment>`, with `codeName` is matched to `<product>` and `versionPrefix` is anything comes before `/`. I use `codeName` as a generalised name for both _OS_ and _Browser_